books/free-programming-books-subjects.md

```diff
@@ -717,6 +717,7 @@ Kerridge (PDF) (email address *requested*, not required)
 ### Programming Paradigms
 
 * [Flow based Programming](http://jpaulmorrison.com/fbp/) - J Paul Morrison
+* [Head First Programming](https://doc.lagout.org/programmation/Head%20First%20-%20Programming.pdf) - Paul Barry, David Griffiths (PDF)
```

Comment:
Not an authorized distribution. Please provide official links to the books where anyone can access them for free

Reply:
Read docs/CONTRIBUTING.md for more information